jupyter_server/serverapp.py

```diff
@@ -470,6 +473,10 @@ def init_settings(
             settings["xsrf_cookie_kwargs"] = {"path": base_url}
         return settings
 
+    def page_config_hook(self, handler, page_config):
+        page_config["allow_custom_env_variables"] = self.allow_custom_env_variables
+        return page_config
+
```

Comment:
page_config is a concept specific to JupyterLab. Up until this point, Jupyter Server has been agnostic to it. This hook might need to land in [jupyterlab_server](https://github.com/jupyterlab/jupyterlab_server) instead, which extends jupyter_server and brings the page_config concept to the server.